python/GafferOSLTest/OSLCodeTest.py

```diff
@@ -0,0 +1,259 @@
+
+		for p in oslShader["parameters"].children() :
+			self.assertEqual( repr( p ), repr( oslCode["parameters"][p.getName()] ) )
+
+		for p in oslShader["out"].children() :
+			self.assertEqual( repr( p ), repr( oslCode["out"][p.getName()] ) )
+
+	def testParseError( self ) :
+
+		n = GafferOSL.OSLCode()
+
+		cs = GafferTest.CapturingSlot( n.errorSignal() )
+		self.assertRaises( RuntimeError, n["code"].setValue, "oops" )
+		self.assertEqual( len( cs ), 1 )
+
+	def testParseErrorDoesntDestroyExistingPlugs( self ) :
+
+		n = GafferOSL.OSLCode()
+		n["parameters"]["in"] = Gaffer.IntPlug( flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+		n["out"]["out"] = Gaffer.IntPlug( direction = Gaffer.Plug.Direction.Out, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+		originalPlugs = n["parameters"].children() + n["out"].children()
+
+	 	self.assertRaises( RuntimeError, n["code"].setValue, "oops" )
+
+	 	self.assertEqual( n["parameters"].children() + n["out"].children(), originalPlugs )
+
+	def testEmpty( self ) :
+
+		# We want empty shaders to still output a
+		# shader so that the ShaderView picks it
+		# up, ready to update when an output is
+		# added.
+
+		n = GafferOSL.OSLCode()
+		self.assertTrue( self._osoFileName( n ) )
+		self.assertEqual( n["type"].getValue(), "osl:shader" )
+
+		n["code"].setValue( "//" )
+		self.assertTrue( self._osoFileName( n ) )
+		self.assertEqual( n["type"].getValue(), "osl:shader" )
+
+		n["code"].setValue( "" )
+		self.assertTrue( self._osoFileName( n ) )
+		self.assertEqual( n["type"].getValue(), "osl:shader" )
+
+	def testMissingSemiColon( self ) :
+
+		n = GafferOSL.OSLCode()
+		n["parameters"]["in"] = Gaffer.IntPlug( flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+		n["out"]["out"] = Gaffer.IntPlug( direction = Gaffer.Plug.Direction.Out, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+
+		# The OSLCode node will often be used to throw in a one-liner,
+		# and omitting a semicolon is an easy mistake that we should
+		# correct automatically.
+		n["code"].setValue( "out = in * 2;" )
+
+	def testAddingAndRemovingPlugsUpdatesShader( self ) :
+
+		oslCode = GafferOSL.OSLCode()
+		oslCode["parameters"]["in"] = Gaffer.IntPlug( flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+		oslCode["out"]["out"] = Gaffer.IntPlug( direction = Gaffer.Plug.Direction.Out, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+
+		oslShader = GafferOSL.OSLShader()
+		oslShader.loadShader( self._osoFileName( oslCode ) )
+		self.assertTrue( "in" in oslShader["parameters"] )
+		self.assertTrue( "out" in oslShader["out"] )
+
+	def testObjectProcessingFunctions( self ) :
+
+		oslCode = GafferOSL.OSLCode()
+		oslCode["out"]["out"] = Gaffer.FloatPlug( direction = Gaffer.Plug.Direction.Out, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+		oslCode["code"].setValue( 'out = inFloat( "s", 0 );' )
```

Comment:
It would have until I added a5905a7. I've updated to use a more explicit `__assertNoError()` method in  d5dc351.